### ISSUE TITLE
Keep Oldest??

### ISSUE THREAD

**Issue by lazarillo** [restic/restic#3383]
Output of `restic version`
--------------------------

`restic 0.12.0 compiled with go1.15.7 on darwin/amd64`

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

See below.

What are you trying to do? What problem would this solve?
---------------------------------------------------------

What I want is perhaps already possible and I am not aware.  But I would like to be able to set a plan that will always save the oldest snapshot, along with all of the other snapshots.

Here is my use case:

- I have a `restic forget --keep-monthly 6 --keep-yearly 6` "policy"
- I like all of it, *except* that I want to also be able to keep the absolute oldest snapshot as well.
  - This snapshot has some data on it that I _think_ I will not need, but I am not 100% sure.  I was confident enough to remove it from my machine, but not from the archive.
- Right now, my oldest is 2021-03-31, but once September rolls around, this will be lost, since `--keep-yearly` will hang on to the most _recent_ snapshot of 2021.

Is there a way I can construct my policy to do exactly what it currently does _plus hang on to the absolute oldest snapshot_?

I saw this mentioned on [issue 2747](https://github.com/restic/restic/issues/2747), but it was not the main point of the discussion.  There might be documentation already explaining how to do it, but I could not find it.

If it does not exist yet, I am happy to help in whatever way I can, but I am not very familiar with Go.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Yes, `restic` is great.  I've been using it for my own personal data storage (old movies and photos) for years.  I am now trying to do this for my extended family so that we have a shared remote backup storage, as backup providers like Google and Dropbox are changing policies and pricing.

**Comment by rawtaz** [comment]
> I would like to be able to set a plan that will always save the oldest snapshot, along with all of the other snapshots.

This is very simple. To do this, simply don't `forget` any snapshots at all.

---

To keep that specific snapshot which you are unsure whether you need or not, just add a tag to id and then use `--keep-tag` with `forget` to make sure it stays.

**Comment by lazarillo** [comment]
Thank you, @rawtaz !

Yes, I can confirm that it works.  To clarify, for any future visitors, I did the following:

```bash
> restic tag --set Original <snapshot_hash>

> restic forget --keep-monthly 6 --keep-yearly 6 --keep-tag Original --prune
```

This does the following:
- Adds the tag `Original` to the snapshot that I wish to keep indefinitely.
- Adds a further rule such that the snapshot(s) with the `Original` tag is saved.

NOTE:  The tag can also be added upon backup creation, but that is a slightly different process.  To add a tag upon backup creation, [see here](https://restic.readthedocs.io/en/stable/040_backup.html#tags-for-backup).  And [see here](https://restic.readthedocs.io/en/latest/manual_rest.html#manage-tags) for managing tags and [here](https://restic.readthedocs.io/en/stable/060_forget.html?highlight=%22keep-tag%22#removing-snapshots-according-to-a-policy) for documentation on how to remember tagged snapshots.

-----

And, of course, when I wrote

> I would like to be able to set a plan that will always save the oldest snapshot, along with all of the other snapshots.

I obviously meant _...along with all of the other snapshots **I am already keeping in my policy**_.  ;-)
